resource-watch/wri-api-components
# Improve the bundle size (depends on #85)

## Overview

This PR updates how the library is bundled and improve the bundle size (-55.7% unminified). It depends on #85.

<p align="center">
<img alt="Schema of the bundling process" src="https://user-images.githubusercontent.com/6073968/106276913-8d1b4f80-6238-11eb-88c6-159b8eb8d8c0.png" />
</p>

The most relevant changes are:
- Bundle the library with [Rollup](https://rollupjs.org/guide/en/) instead of Webpack
- Added unbundled ES modules (allowing for tree-shaking)
- Replaced [Moment](https://momentjs.com/) by [dayjs](https://day.js.org/)
- Removed dependency to [layer-manager](https://github.com/Vizzuality/layer-manager/)
- Removed unused icons (⚠️ **make sure to review these changes**)

## Testing instructions

1. Make sure the components are still working in Storybook: `yarn start`
2. Make sure the library can be bundled: `yarn build`
3. You can test the library in a project by building it and linking it with [yalc](https://github.com/wclr/yalc) (`yarn link` doesn't fully work)

## Pivotal task

Not tracked.
